pkg/component/observability/opentelemetry/collector/collector.go

```diff
@@ -85,12 +144,122 @@ func (o *otelCollector) WaitCleanup(ctx context.Context) error {
 	return managedresources.WaitUntilDeleted(timeoutCtx, o.client, o.namespace, managedResourceName)
 }
 
-func (o *otelCollector) openTelemetryCollector(namespace, lokiEndpoint string) *otelv1beta1.OpenTelemetryCollector {
-	return &otelv1beta1.OpenTelemetryCollector{
+func (o *otelCollector) emptyScrapeConfig() *monitoringv1alpha1.ScrapeConfig {
+	return &monitoringv1alpha1.ScrapeConfig{ObjectMeta: monitoringutils.ConfigObjectMeta(scrapeConfigName, o.namespace, shoot.Label)}
+}
+
+func (o *otelCollector) scrapeConfig() (*monitoringv1alpha1.ScrapeConfig, error) {
+	allowedMetrics := []string{
+		"otelcol_exporter_enqueue_failed_log_records",
+		"otelcol_exporter_enqueue_failed_metric_points",
+		"otelcol_exporter_enqueue_failed_spans",
+		"otelcol_exporter_queue_capacity",
+		"otelcol_exporter_queue_size",
+		"otelcol_exporter_send_failed_log_records",
+		"otelcol_exporter_send_failed_metric_points",
+		"otelcol_exporter_send_failed_spans",
+		"otelcol_exporter_sent_log_records",
+		"otelcol_exporter_sent_metric_points",
+		"otelcol_exporter_sent_spans",
+		"otelcol_process_cpu_seconds",
+		"otelcol_process_memory_rss",
+		"otelcol_process_runtime_heap_alloc_bytes",
+		"otelcol_process_runtime_total_alloc_bytes",
+		"otelcol_process_runtime_total_sys_memory_bytes",
+		"otelcol_process_uptime",
+		"otelcol_processor_incoming_items",
+		"otelcol_processor_outgoing_items",
+		"otelcol_receiver_accepted_log_records",
+		"otelcol_receiver_accepted_metric_points",
+		"otelcol_receiver_accepted_spans",
+		"otelcol_receiver_refused_log_records",
+		"otelcol_receiver_refused_metric_points",
+		"otelcol_receiver_refused_spans",
+		"otelcol_scraper_errored_metric_points",
+		"otelcol_scraper_scraped_metric_points",
+	}
+
+	scrapeConfig := o.emptyScrapeConfig()
+	_, err := controllerutils.GetAndCreateOrMergePatch(context.TODO(), o.client, scrapeConfig, func() error {
```

Comment:
Pass a context in the function arguments and replace `context.TODO()`

```suggestion
	_, err := controllerutils.GetAndCreateOrMergePatch(ctx, o.client, scrapeConfig, func() error {
```